Графика для всех страниц

tayga2410 opened the pull request:
---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2325583/tasks/5)

---
:mortar_board: [Графика для всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2325583/tasks/7)

tayga2410 committed:
html changes

tayga2410 committed:
1st page of sedona

tayga2410 committed:
index and catalog fixed

dum-man commented on index.html:
```diff
@@ -9,17 +9,35 @@
+  <header class="page-header">
 
 <nav class="navigation">
     <img src="img/logo.svg" alt="логотип">
```
Изображению нужно прописать класс, например `header-logo`, он понадобится для будущей стилизации и лучше обернуть изображение в ссылку (тег `<a>`) с классом `navigation-logo-link`, чтобы по клику на логотип мы могли переходить на главную страницу, и так как мы уже находимся на главной странице - не нужно ему прописывать путь (атрибут `href`)

dum-man commented on index.html:
```diff
@@ -9,17 +9,35 @@
+        <li><a class="navigation-item" href="catalog.html">Гостиницы</a>
+        </li>
+
+      </ul>
```
Класс `navigation-item` лучше прописать на элементы списка `li`, так как именно они являются дочерними по отношения к самому списку `ul`, а на сами ссылки прописать класс `navigation-link`. На первую ссылку с содержанием "Главная" нужно добавить класс, например `navigation-link-current` указывающий, что мы находимся по адресу данной ссылки, мы потом с помощью этого класса через стили сможем визуально его выделить.

dum-man commented on index.html:
```diff
@@ -9,17 +9,35 @@
+        <li class="navigation-item">
+          <a class="navigation-link" href="#">ХОЧУ СЮДА!</a>
+        </li>
+      </ul>
```
Данному списку уже не нужно указывать класс `navigation-list`, нужно оставить только класс `navigation-user`, чтобы на этот список у нас не применялись нежелательные стили. Элементам списка, а также ссылкам нужно указать класс с добавлением слова `user`, например элементам списка: `navigation-user-item`, ссылкам: `navigation-user-link`, так как все вложенные элементы относятся именно к списку `navigation-user`.

dum-man commented on index.html:
```diff
@@ -9,17 +9,35 @@
+        </li>
+        <li class="navigation-item">
+          <a class="navigation-link" href="#">
+            <span class="visually-hidden">Лайк</span>
```
Вместо "Лайк" лучше укажи "Избранное", так будет более понятно куда ведет эта ссылка. Также если внимательно присмотреться, то у нас на иконке "Избранное" есть небольшой бэйдж с количеством элементов находящихся в избранном. Сюда будет уместно добавить тег, который не имеет семантической нагрузки, например `span`, добавить содержание как по макету: `12` и указать ему класс, например `notification-badge`.

dum-man commented on index.html:
```diff
@@ -31,73 +49,124 @@
       <p class="grand-canyon">Because the Grand Canyon sucks!</p>
 
         </section>
```
В данном случае мы отобразим весь текст при помощи добавления изображении, такой вариант будет гораздо оптимальнее. Добавь сюда два тега `img`, но при этом все равно текстовый заголовок должен присутствовать, так как это блок описывающий всю страницу, добавь сюда заголовок первого уровня с подходящим описанием, но скрой его классом `visually-hidden`.

dum-man commented on index.html:
```diff
@@ -31,73 +49,124 @@
+
+    </li>
+  </ul>
+</section>
```
Каждая секция у нас должна иметь свой заголовок, даже если он не указан на макете, то есть если есть тег `section`, то внутри него обязательно должен находится заголовок описывающий его содержание, на такие заголовки мы обычно указываем класс `visually-hidden`, чтобы скрыть его со страницы . Заголовок должен следовать иерархии заголовков, например в данном случае будет правильно указать заголовок второго уровня с содержанием описывающим данную секцию, например "Преимущества". Заголовки которые ты указал в этой секции на самом деле не являются заголовками, они являются простым текстом `p`.

dum-man commented on index.html:
```diff
@@ -31,73 +49,124 @@
+  <h1 class ="advantages-header">Седона-небольшой городок в аризоне, заслуживающий большего!</h1>
+  <h3 class ="advantages-bottom">рассмотрим 5 причин, по которым Седона круче, чем Гранд-Каньон!</h3>
+  <ul class="Advantages-list">
+    <li class="Advantages-item">
```
Названия классов должны начинаться с маленькой буквы

dum-man commented on index.html:
```diff
@@ -31,73 +49,124 @@
+    <p class="advantages-description">Да, по нему можно пройти! Если вы осмелитесь, разумеетя</p>
+
+  </li>
+</ul>
```
Все эти отдельные блоки `ul` должны быть частью самого первого списка `advantages-list`, не нужно для каждого элемента списка `li` создавать отдельный `ul`, все они должны быть дочерними по отношению к первому списку в этом блоке.  По макету раздел преимущества включает в себя все эти пять элементов, соотвественно они должны быть частью  одного `ul`

dum-man commented on index.html:
```diff
@@ -31,73 +49,124 @@
+    <p class="advantages-description">Не только китайского, но и настоящего местного производства</p>
+
+  </li>
+</ul>
```
Здесь пропущен блок делящий страницу на секции, то есть тег `section`. Заголовок первого уровня на странице может быть только один, мы его указали в блоке "hero", здесь нужно добавить скрытый заголовок второго уровня с содержанием вкратце описывающий содержание всей секции, например "Инфраструктура", и соответсвенно в названиях классов поменять `advantages` на `infrastructure`. С разделением каждого элемента списка на отельный список ситуация такая же как и выше, нужно завести один список `ul` и все элементы списка `li` вложить в него.

dum-man commented on index.html:
```diff
@@ -31,73 +49,124 @@
       Укажите предполагаемые даты поездки, <br>
       и мы покажем вам лучшие предложения гостиниц в Седоне<br>
       <button type="submit">Поиск гостиницы в Седоне</button>
     </div>
```
На данном этапе можно обойтись без использования тега `div`, нужно обернуть этот блок в тег `section`, так как эта часть по сути тоже является отдельной секцией. Класс можно указать просто `search`, так как у нас на странице всего один поиск. Текст расположенный после заголовка нужно обернуть в параграф (тег `<p>`), от тега `<br>` нужно избавиться, его использования не рекомендуется, текст мы буде переносить на следующую строчку при помощи стилей. Вместо кнопки `button` лучше указать ссылку и добавить подходящий по смыслу класс.

dum-man commented on index.html:
```diff
@@ -31,73 +49,124 @@
+      <h1>Подпишитесь на рассылку</h1>
+      <h3 class="subscribe-info">Только полезная информация и никакого спама,
+        честное бойскаутское!</h3>
+    </section>
```
От тега `img` в этой секции можно избавиться, изображение мы добавим скорее всего при помощи стилей. Также насчет заголовка первого уровня, на странице он может быть только один, здесь нужно его заменить на заголовок второго уровня. Текст обернутый в заголовок третьего уровня нужно обернуть просто в параграф, по смыслу этот текст не является заголовком. В эту секцию нужно добавить форму тег `form` и как на макете добавить одно поле ввода тег `input` и одну кнопку.

dum-man commented on index.html:
```diff
@@ -31,73 +49,124 @@
+    </section>
 
 
   </footer>
```
Внутри тега `footer` не нужна дополнительная секция, футер сам по себе уже является секцией, так что можно обойтись без тега `section`. Вместо разделения иконок на отдельные секции, нужно для них создать список и сделать все иконки элементам этого списка. Заголовки тут не нужны, нужно внутри каждой ссылки добавить `span` с классом `visually-hidden` и текст который указан в этих заголовках - перенести в этот спан.

tayga2410 committed:
catalog end, checked via validator

tayga2410 reviewed (commented):
checked via validator

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+        <h1 class="header-catalog-title">Гостиницы Седоны</h1>
+        <ul class="breacrumbs">
+          <li class="breadcrumbs-item">
+            <a class="breadcrumbs-link" href="index.html">Главная</a>
```
Внутри ссылки текст "Главная" нужно обернуть в `span` с классом `visually-hidden`, так как по следуя макету мы добавим сюда иконку

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+            <a class="breadcrumbs-link">Гостиницы</a>
+          </li>
+        </ul>
       </header>
```
Все содержимое блока фильтрации нужно вложить в `header`, то есть `header` у тебя закроется перед секцией с заголовком "гостиниц найдено:38".

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
-          </a>
-        </section>
+      <section class="hotel-filter">
+        <h3 class="hotel-filter-title">Инфраструктура:</h3>
```
Здесь секция не нужна, отдельные блоки фильтрации можно группировать с помощью тега   `fieldset` и вместо заголовка `h3` для обозначения заголовка блока нужно использовать тег `legend`

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+        <h3 class="hotel-filter-title">Инфраструктура:</h3>
+        <ul class="hotel-filter-list">
+          <li class="hotel-filter-item">
+            <label> <input type="checkbox" checked>Бассейн </label>
```
Во всех трех инпутах (Бассейн, Парковка, Wi-Fi) пропущен атрибут `name`, он будет нужен для передачи выбранного значения на сервер. Можешь дать им имя отталкиваясь от содержимого этих инпутов, например в первом инпуте `name="swimming-pool"`

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+        </ul>
+
+        <!-- поскольку по смыслу эти 2 бокса одинаковые. то я посчитал нужным оставить классы как есть,
+  возможно во 2 боксе нужно поменять класс на class=hotel-filter-type, поскольку это тип жилья-->
```
Второй блок фильтрации "Тип жилья:" также нужно сгруппировать с помощью тега `fieldset` и вместо заголовка h3 для обозначения заголовка блока нужно использовать тег `legend`

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+        <h3 class="hotel-filter-title">Тип жилья:</h3>
+        <ul class="hotel-filter-list">
+          <li class="hotel-filter-item">
+            <label> <input type="radio" checked>Гостиница </label>
```
Тут также во всех трех инпутах (Гостиница, Мотель, Апартаменты) пропущен атрибут `name`, но так как эти инпуты типа `radio`, то значения этого атрибута должно быть для всех одинаковым, можешь также при выборе имени отталкиваться от предназначения этих инпутов и дать им имя что нибудь на подобии `housing-type`, но также обрати внимание, что для корректной отправки выбранного значения на сервер у этих инпутов должен иметься атрибут `value` со своим уникальным значением

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+        </ul>
+
+        <div class="hotel-filter-list">
+          <h3 class="hotel-filter-title">Стоимость, ₽</h3>
```
Также вместо `div` используй тег `fieldset` и вместо заголовка `h3` для обозначения заголовка блока нужно использовать тег `legend`

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+          <h3 class="hotel-filter-title">Стоимость, ₽</h3>
+          <p class="price-range"></p>
+          <p class="min-price"></p>
+          <p class="visually-hidden">Min Price</p>
```
Строчка с первым тегом `p` тут излишняя. Остальные два тега `p` замени на менее смысловые теги `span` и текст внутри тегов `visually-hidden` пиши на русском языке

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+          <input type="text" id="min-value" placeholder="От">
+
+          <p class="max-price"></p>
+          <p class="visually-hidden">Max Price</p>
```
Тут также два тега p замени на менее смысловые теги span и текст внутри тегов visually-hidden должен быть на русском языке

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+
+          <!--сомневаюсь насчет привязки кнопок к price range-->
+          <button class="price-apply" type="submit">применить</button>
+          <button class="price-reset" type="reset">сбросить</button>
```
Названия классов для кнопок можешь уточнить и добавить им слово `button`

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+      </section>
+
+      <section>
+        <h2 class="hotel-find">гостиниц найдено:38</h2>
```
Текст "гостиниц найдено:38" не является заголовком, так как не резюмирует содержание секции, здесь правильно будет перенести этот текст в тег `p`, а количество найденных гостиниц, то есть цифру "38" перенести в тег `span` так как содержание этого тега будет динамически меняться. А для заголовка добавить содержание "Список гостиниц" и не забыть про класс `visually-hidden`, так как по макету у нас заголовка нет

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+      <section>
+        <h2 class="hotel-find">гостиниц найдено:38</h2>
+        <select class="hotel-options" aria-label="Сортировка">
+          <option value="cheap">сначала дешевые</option>
```
Хотя и по макету этого не видно, но можно предположить, что у этого выпадающего селекта будут еще некоторые опции, давай добавим еще два раза `option` с содержанием "Сначала популярные" и "Сначала дорогие", также не забудь указать верное значение атрибуту `value`

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+              <img class="hotel-card-image" src="#" alt="Amara Resort & Spa"></a>
+            <h3 class="hotel-card-title">Amara Resort & Spa</h3>
+            <p class="hotel-card-description">Гостиница от 4000 ₽</p>
+            <a class="hotel-card-about" href="#">Подробнее</a>
+            <a class="hotel-card-favourite" href="#">В избранное</a>
```
Первая ссылка "Подробнее" указана верно, так как по клику на нее будет открыта страница соответствующей гостиницы, но вот что касается второй ссылки, по клику на нее у нас в список избранного будет добавляться новый элемент, так что правильнее будет здесь сделать кнопку с `type="button"`

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+            <p class="hotel-card-description">Гостиница от 4000 ₽</p>
+            <a class="hotel-card-about" href="#">Подробнее</a>
+            <a class="hotel-card-favourite" href="#">В избранное</a>
+            <img class="hotel-card-stars" src="#" alt="Рейтинг отеля">
```
Картинку поменяй на тег `span`, звездочки мы будет добавлять через стили так как это декоративное изображение, а не контентное, тег `img` здесь не нужен

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+            <a class="hotel-card-about" href="#">Подробнее</a>
+            <a class="hotel-card-favourite" href="#">В избранное</a>
+            <img class="hotel-card-stars" src="#" alt="Рейтинг отеля">
+            <a class="hotel-card-rating" href="#">Рейтинг</a>
```
Рейтинг визуально похож на ссылку или кнопку, но на самом деле таковым не является, это просто обычный тег, пока можешь сделать его через `span` и добавь соответствующее значение рейтинга как на макете
Повтори все изменения касающиеся карточки отеля для каждого элемента списка

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+            <a class="hotel-card-rating" href="#">Рейтинг: 5,0</a>
+          </li>
+        </ul>
+        <ul>
```
У открывающего тега `ul` пропущено название класса

dum-man commented on catalog.html:
```diff
@@ -1,138 +1,242 @@
+        </div>
+      </section>
+
+      <section>
```
У `section` пропущено название класса

tayga2410 committed:
code fixes + image input

Переделал код, вставил картинки. Некоторые я так понял не нужны, потому что их потом через css вставлять, но всеравно вставил.
Картинки прогнал через Squoosh и svgomg

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
-      <header class="header-catalog">
-        <h1 class="header-catalog-title">Гостиницы Седоны</h1>
+      <section class="hotel-filter">
+        <img class="hotel-filter-background" src="images/catalog-background.jpg" width="1200" height="412">
```
Размытый фон в этом блоке будем добавлять через стили, пока эту картинку можешь удалить

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
           <li class="breadcrumbs-item">
-            <a class="breadcrumbs-link" href="index.html">Главная</a>
+            <a class="breadcrumbs-link" href="index.html">
+              <img class="breadcrumbs-link-home" src="images/icon-home.svg" width="12" height="12" alt="на главную">
```
Иконку добавим через стили, пока можешь ее удалить и на ее место добавить `span` с классом `visually-hidden` с текстом "Главная"

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
-
-      <section class="hotel-filter">
-        <h3 class="hotel-filter-title">Инфраструктура:</h3>
+<fieldset class="hotel-filter-box">
```
До открывающего тега `fieldset` нужно добавить форму, добавь тег `form` с атрибутами `action="https://echo.htmlacademy.ru/" method="get"` и не забудь про соответствующий класс, форма закроется перед первым закрывающемся тегом `section`

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
+            <label> <input type="checkbox" name="wireless-internet">Wi-Fi </label>
           </li>
         </ul>
 
```
первый тег `fieldset` должен закрыться здесь. Следуя макету все три блока формы, а именно: первый блок "Инфраструктура:", второй блок "Тип жилья:" и трети блок "Стоимость, ₽:" должны быть обернуты в отдельные блоки `fieldset` и каждого такого блока должен быть свой тег `legend` с текстом который является заголовком для этих блоков

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
 
-        <!-- поскольку по смыслу эти 2 бокса одинаковые. то я посчитал нужным оставить классы как есть,
-  возможно во 2 боксе нужно поменять класс на class=hotel-filter-type, поскольку это тип жилья-->
-        <h3 class="hotel-filter-title">Тип жилья:</h3>
```
Здесь должен начаться второй блок `fieldset`

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
+            <label> <input type="radio" name="hotel-type" value="Apartments">Апартаменты </label>
           </li>
         </ul>
 
```
Здесь второй блок `fieldset` должен закрыться

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
           </li>
         </ul>
 
         <div class="hotel-filter-list">
```
Вместо дива сделай трети `fieldset`

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
+          <legend class="hotel-filter-title">Стоимость, ₽</legend>
+          <span class="min-price"></span>
+          <span class="visually-hidden">Минимальная цена</span>
           <input type="text" id="min-value" placeholder="От">
```
Тип инпута можешь поменять на `type="number"`, `span` и `input` оберни в `label`

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
-          <p class="visually-hidden">Max Price</p>
+          <span class="max-price"></span>
+          <span class="visually-hidden">Максимальная цена</span>
           <input type="text" id="max-value" placeholder="До">
```
Тип инпута можешь поменять на `type="number"`, `span` и `input` оберни в `label`

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
+          <span class="max-price"></span>
+          <span class="visually-hidden">Максимальная цена</span>
           <input type="text" id="max-value" placeholder="До">
           <input type="range" min="0" max="9000" step="1" value="50">
```
Оберни этот `input` в `label` и добавь `span` с классом `visually-hidden` с текстом "Ценовой диапазон"

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
 
-      <section>
-        <h2 class="hotel-find">гостиниц найдено:38</h2>
+      <section class="hotel-name">
```
Название для класса будет уместнее дать что нибудь на подобии `hotels` так как эта секция будет представлять все отели

dum-man commented on catalog.html:
```diff
@@ -40,130 +43,139 @@
           <span class="visually-hidden">показать списком</span>
         </a>
         <ul>
+          <h3 class="visually-hidden">Список гостиниц</h3>
```
Этот заголовок поменяй на `h2` и перенеси в начало секции

tayga2410 committed:
Update catalog.html

dum-man commented on catalog.html:
```diff
@@ -44,18 +44,16 @@
-            <a class="breadcrumbs-link" href="index.html">
-              <img class="breadcrumbs-link-home" src="images/icon-home.svg" width="12" height="12" alt="на главную">
-            </a>
+            <a class="breadcrumbs-link" href="index.html"></a>
```
Внутрь этой ссылки добавь `span` с классом `visually-hidden`  и текстом "Главная", так как по макету вместо текста у нас будет только иконка

dum-man commented on catalog.html:
```diff
@@ -86,26 +86,33 @@ <h1 class="hotel-filter-title">Гостиницы Седоны</h1>
+        </label>
 
           <button class="price-apply-button" type="submit">применить</button>
           <button class="price-reset-button" type="reset">сбросить</button>
```
Эти две кнопки вынеси за пределы `fieldset`, но оставь внутри самой формы

dum-man commented on catalog.html:
```diff
@@ -86,26 +86,33 @@ <h1 class="hotel-filter-title">Гостиницы Седоны</h1>
-          <legend class="hotel-filter-title">Стоимость, ₽</legend>
+      <fieldset class="hotel-filter-box">
+        <legend class="hotel-filter-title">Стоимость, ₽</legend>
           <span class="min-price"></span>
```
Добавь внутрь этого спана текст "от"

dum-man commented on catalog.html:
```diff
@@ -86,26 +86,33 @@ <h1 class="hotel-filter-title">Гостиницы Седоны</h1>
           <span class="visually-hidden">Минимальная цена</span>
           <input type="text" id="min-value" placeholder="От">
 
           <span class="max-price"></span>
```
А внутрь этого спана добавь текст "от"

dum-man commented on catalog.html:
```diff
@@ -86,26 +86,33 @@ <h1 class="hotel-filter-title">Гостиницы Седоны</h1>
+      <fieldset class="hotel-filter-box">
+        <legend class="hotel-filter-title">Стоимость, ₽</legend>
           <span class="min-price"></span>
           <span class="visually-hidden">Минимальная цена</span>
           <input type="text" id="min-value" placeholder="От">
```
Оберни этот спан и инпут в отдельный тег `label`

tayga2410 committed:
fix catalog 05.02.2023 16:53
